docs/extensions/installer-extensions.md

```diff
@@ -0,0 +1,217 @@
+The currently supported arguments are:
+
+- `kind` (if missing, defaults to `pod`)
+- `namespace`
+- `condition` (defaults to `ready`)
+- `timeout` (in seconds; defaults to 300)
+- `selector`. If the value of `selector` is `all`, FuseML will wait for all resources of given `kind` to reach the indicated `condition`. Otherwise, the `selector` value is treated like the value for the `--selector` option for the `kubectl wait` command.
+
+```yaml
+[...]
+services:
+  - id: API
+    resource: kserve-api
+    category: prediction-serving
+    description: kserve prediction service API
+    endpoints:
+      - url: http://kubernetes.default.svc
+        type: internal
+  - id: UI
+    resource: kserve-ui
+    category: UI
+    description: KServe UI
+[...]
+```
+
+The `services` section provides information that is reflected in the generated Extension Descriptor that FuseML will use to register the extension in the [FuseML Extension Registry](extension-registry.md), more specifically in the `services` field of the Extension Record. For more information about the fields that can be configured in the services section, see [Extension Descriptor](extension-registry.md#extension-records).
+
+!!! note
+
+    The `services` field should only be configured for extensions that expose APIs that need to be accessed from the FuseML workflows. Extensions that are not meant to interact with FuseML workflows (e.g. those that are only used as functional dependencies for other extensions) do not need to have this field filled in and will be ignored by FuseML in relation with workflows.
+
+```yaml
+[...]
+gateways:
+  - name: kserve-web-app
+    namespace: kserve
+    servicehost: kserve-models-web-app
+    port: 80
+[...]
+```
+
+Some Kubernetes applications such as KServe don't include means of exposing the provided services in their official deployment manifests (e.g. Ingress, Load Balancer or Istio Gateway). When the `gateways` field is specified in the descriptor, the FuseML installer will also create one Istio gateway corresponding to each entry.
+
+```yaml
+[...]
+rolerules:
+  - apigroups:
+      - serving.kserve.io
+    resources:
+      - inferenceservices
+    verbs:
+      - get
+      - list
+      - create
+      - patch
+      - watch
+```
+
+Some third party AI/ML services, like KServe, come in the form of Kubernetes operators that have associated CRD. To have access to these resources, the workflow steps need to have additional permissions granted to them. The `rolerules` section can be used to provide the list of permissions (Kubernetes roles) that need to be granted to FuseML workflow steps that need to access the extension. The syntax is the same as for the [Kubernetes Role object](https://kubernetes.io/docs/reference/access-authn-authz/rbac/), i.e. it is expected to contain ApiGroups, Resources and Verbs values.
```

Comment:
It would be probably right to mention the specific role that is being affected by this: _fuseml-workloads_ under _fuseml-workloads_ namespace

Reply:
Great point. I added a note detailing this and instructing users to manually add permissions to that service account if they don't / can't model them as `rolerules`.